accepted/2024/support-nuget-authentication-plugins-dotnet-tools.md

```diff
@@ -0,0 +1,264 @@
+# ***Support for NuGet authentication plugins deployed via .NET tools***
+
+- Author Name: https://github.com/kartheekp-ms
+- GitHub Issue: https://github.com/NuGet/Home/issues/12567
+
+## Summary
+
+Currently, NuGet utilizes a [cross-platform plugin model](https://learn.microsoft.com/nuget/reference/extensibility/nuget-cross-platform-plugins#supported-operations) which is primarily used for [authentication against private feeds](https://learn.microsoft.com/nuget/reference/extensibility/nuget-cross-platform-authentication-plugin).
+It also supports the [package download](https://github.com/NuGet/Home/wiki/NuGet-Package-Download-Plugin) operation.
+
+To accommodate all scenarios involving NuGet client tools, plugin authors will need to create plugins for both `.NET Framework` and `.NET Core`.
+The following details the combinations of client and framework for these plugins.
+
+| Client tool | Framework |
+|-------------|-----------|
+| Visual Studio | .NET Framework |
+| dotnet.exe | .NET Core |
+| NuGet.exe | .NET Framework |
+| MSBuild.exe | .NET Framework |
+| NuGet.exe on Mono | .NET Framework |
+
+Currently, NuGet maintains `netfx` folder for plugins that will be invoked in `.NET Framework` code paths, `netcore` folder for plugins that will be invoked in `.NET Core` code paths.
+The proposal is to add a new `tools` folder to store NuGet plugins that are deployed as [tool Path](https://learn.microsoft.com/dotnet/core/tools/global-tools-how-to-use#use-the-tool-as-a-global-tool-installed-in-a-custom-location) .NET tools.
+
+| Framework | Root discovery location |
+|-----------|------------------------|
+| .NET Core | %UserProfile%/.nuget/plugins/netcore |
+| .NET Framework | %UserProfile%/.nuget/plugins/netfx |
+| .NET Framework & .NET Core [current proposal] |  %UserProfile%/.nuget/plugins/tools |
+
+## Motivation 
+
+At present, NuGet uses different methods to execute plugins for `.NET Framework` and `.NET Core`.
+For the `.NET Framework`, it looks for files ending in `*.exe`, while for `.NET Core`, it looks for files ending in `*.dll`.
+These files are usually stored in two separate folders under the base path for NuGet plugins.
+
+Before `.NET Core 2.0`, this division was necessary because `.NET Core` only supported platform-agnostic DLLs.
+However, with the latest versions of .NET, this division is no longer necessary, but the NuGet plugin architecture still requires supporting and deploying multiple versions.
```

Comment:
I think this is misleading. 

The reason why we have net core and net fx is because when you have a netfx process, you're not guaranteed to have the .NET runtime installed. 
Nothing in the latest versions has changed that. 

What has changed is that .exe is supported now. That's stylistic and easy to make work if want to for .NET plugins.

Reply:
Thank you, Nikolche, for the feedback. Andy also provided similar feedback in [this GitHub discussion](https://github.com/NuGet/Home/pull/13242#discussion_r1511935960), and I have updated the spec accordingly. Please review it again, and let me know if the new wording in this section addresses your feedback.

Reply:
Given that I have addressed your feedback, I am marking this conversation as resolved. Feel free to unresolve this conversation if you other feedback regarding the wording.